rosidl_generator_py/cmake/rosidl_generator_py_generate_interfaces.cmake

```diff
@@ -68,15 +68,15 @@ foreach(_idl_file ${rosidl_generate_interfaces_IDL_FILES})
   endif()
 endforeach()
 
+file(MAKE_DIRECTORY "${_output_path}")
+file(WRITE "${_output_path}/__init__.py" "")
+
 if(NOT "${_generated_msg_py_files} " STREQUAL " ")
   list(GET _generated_msg_py_files 0 _msg_file)
   get_filename_component(_parent_folder "${_msg_file}" DIRECTORY)
   list(APPEND _generated_msg_py_files
     "${_parent_folder}/__init__.py"
   )
-  # TODO(dirk-thomas) since this are being generated at configure time
-  # it should not be a dependency of the custom command
-  file(WRITE "${_parent_folder}/__init__.py" "")
```

Comment:
it's generated [here](https://github.com/ros2/rosidl/blob/master/rosidl_generator_py/rosidl_generator_py/generate_py_impl.py#L98). IMO if we got the problem previously is because we were not constraining the scope to the msg folder so it was expecting an __init__.py file in the srv folder and could never find it.